d2graph/d2graph.go

```diff
@@ -1052,6 +1052,45 @@ func (obj *Object) GetDefaultSize(mtexts []*d2target.MText, ruler *textmeasure.R
 	return &dims, nil
 }
 
+// resizes the object to fit content of the given width and height in its inner box with the given padding.
+// this accounts for the shape of the object, and if there is a desired width or height set for the object
+func (obj *Object) SizeToContent(contentWidth, contentHeight, paddingX, paddingY float64) {
+	var desiredWidth int
+	var desiredHeight int
+	if obj.WidthAttr != nil {
+		desiredWidth, _ = strconv.Atoi(obj.WidthAttr.Value)
+	}
+	if obj.HeightAttr != nil {
+		desiredHeight, _ = strconv.Atoi(obj.HeightAttr.Value)
+	}
+
+	dslShape := strings.ToLower(obj.Shape.Value)
+	shapeType := d2target.DSL_SHAPE_TO_SHAPE_TYPE[dslShape]
+	s := shape.NewShape(shapeType, geo.NewBox(geo.NewPoint(0, 0), contentWidth, contentHeight))
+
+	var fitWidth, fitHeight float64
+	if shapeType == shape.PERSON_TYPE {
+		fitWidth = contentWidth + paddingX
+		fitHeight = contentHeight + paddingY
+	} else {
+		fitWidth, fitHeight = s.GetDimensionsToFit(contentWidth, contentHeight, paddingX, paddingY)
+	}
+	obj.Width = math.Max(float64(desiredWidth), fitWidth)
```

Comment:
This is not obeying the set size. Only if it is smaller than the content.
From what I remember, we simply ignore the set size for containers. Maybe, it should be same here. It doesn't make much sense to be different from the current behavior

Reply:
updated tala to use desiredWidth as long as content fits